internal/metrics/otelmetrics/factory.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package otelmetrics
+
+import (
+	"context"
+	"log"
+
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	metric "go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+type otelFactory struct {
+	meter metric.Meter
+}
+
+func NewFactory() metrics.Factory {
+	res, err := resource.New(context.Background(),
```

Comment:
the factory is supposed to be an adapter, it's not meant to actually initialize the SDK. When used from OTEL Collector framework the OTEL SDK would already be initialized.

Reply:
Oh, but I have to call the OTEL.setMeterProvider to create a global instance of meter to prevent a no-op implementation being returned whenever the factory calls for Otel.Meter.
So, does that mean a global meter would already be created for the factory to access?

Reply:
You need to pass MeterProvider as an argument to the factory. Where it's created is not the business of the factory.

You do need to create it in the benchmark.

Reply:
Noted, Thanks